src/main/java/org/terasology/launcher/packages/PackageManager.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.util.DirectoryUtils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Handles installation, removal and update of game packages.
+ */
+public class PackageManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(PackageManager.class);
+
+    private final Repository onlineRepository;
+    private LocalRepository localRepository;
+
+    public PackageManager() {
+        onlineRepository = new JenkinsRepository();
+    }
+
+    public void initLocalStorage(Path gameDirectory, Path cacheDirectory) {
+        try {
+            DirectoryUtils.checkDirectory(gameDirectory);
+            DirectoryUtils.checkDirectory(cacheDirectory);
+            localRepository = new LocalRepository(gameDirectory, cacheDirectory);
+            localRepository.loadCache();
+        } catch (IOException e) {
+            logger.error("Error initialising local storage: {}", e.getMessage());
+        }
```

Comment:
I'd argue that all this could be moved to a method in `Cache` (`LocalRepository`) -- probably a static factory method like `Cache.load` that creates a new `Cache` _and_ loads it at the same time. That would simplify the method down to a single line, `cache = Cache.load(gameDirectory, cacheDirectory)`.

Reply:
That does sound like a simpler API, though I usually tend to avoid static methods mostly (been hearing not-so-good things about them all the time so) :sweat_smile: But this goes well with the separation of local storage (original idea of `LocalRepository`) from cache storage (`Cache`).